third_party/move/move-vm/runtime/src/storage/ty_layout_converter.rs

```diff
@@ -103,6 +106,120 @@ where
         self.struct_definition_loader.runtime_environment()
     }
 
+    pub(crate) fn contains_option_type(
+        &self,
+        ty: &Type,
+        gas_meter: &mut impl DependencyGasMeter,
+        traversal_context: &mut TraversalContext,
+    ) -> PartialVMResult<bool> {
+        let option_module_id =
+            ModuleId::new(AccountAddress::ONE, Identifier::from(ident_str!("option")));
+        let option_struct_name = Identifier::from(ident_str!("Option"));
+        use Type::*;
+        match ty {
+            Signer | Bool | Address | U8 | U16 | U32 | U64 | U128 | U256 | TyParam(..) => Ok(false),
+            Reference(ty) | MutableReference(ty) => {
+                self.contains_option_type(ty, gas_meter, traversal_context)
+            },
+            Vector(ty) => self.contains_option_type(ty, gas_meter, traversal_context),
+            Struct { idx, .. } => {
```

Comment:
Technically, `Option` is generic so this should not be possible? I am fine keeping the check just in case.